src/Raven.Client/Documents/Listeners/IDocumentConversionListener.cs

```diff
@@ -0,0 +1,36 @@
+//-----------------------------------------------------------------------
+// <copyright file="IDocumentConversionListener.cs" company="Hibernating Rhinos LTD">
+//     Copyright (c) Hibernating Rhinos LTD. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+using Raven.Json.Linq;
+
+namespace Raven.Client.Documents.Listeners
+{
+    /// <summary>
+    /// Extended hook for users to provide additional logic for converting to / from 
+    /// entities to document/metadata pairs.
+    /// </summary>
+    public interface IDocumentConversionListener
+    {
+        /// <summary>
+        /// Called before converting an entity to a document and metadata
+        /// </summary>
+        void BeforeConversionToDocument(string key, object entity, RavenJObject metadata);
+
+        /// <summary>
+        /// Called after having converted an entity to a document and metadata
+        /// </summary>
+        void AfterConversionToDocument(string key, object entity, RavenJObject document, RavenJObject metadata);
```

Comment:
It's temporary just to make it work.
notice that storeListener and deleteListener are already implemented in the right way